### PR TITLE
added environment tag helper

### PR DESCRIPTION
**Pull request by danroth27**

### CODE REVIEW

**Comment by danroth27** [comment]
@Rick-Anderson @NTaylorMullen @dougbu Please review

File: aspnet/fundamentals/environments.rst [hunk 1]
```diff
@@ -72,6 +72,24 @@ For example, you can use the following code in you Configure method to setup env
 
 If the app is running in a ``Development`` environment, then it enables BrowserLink and development specific error pages (which typically should not be run in production). Otherwise, if the app is not running in a development environment, a standard error handling page is configured to be displayed in response to any unhandled exceptions.
 
+You may need to determine which content to send to the client at runtime, depending on the current environment. A common scenario when this happens is when a developer uses full scripts and style sheets for development and debugging vs. their minified counterparts for staging or production. You can do this by setting the names attribute of the ``<environment>`` Tag Helper that corresponds to the environment you want it to work in - ``Development``, ``Staging`` or ``Production``.
```
Comment (Rick-Anderson) [inline aspnet/fundamentals/environments.rst]: In a development environment, you generally serve non-minimized scripts and style sheets, which makes debugging easier. Production and test environments should be served the minified versions and generally from a CDN. You can ...

**Comment by NTaylorMullen** [comment]
EnvironmentTagHelper is only available via the `Microsoft.AspNetCore.Mvc.TagHelpers` package (and its `@addTagHelper` inclusion). May want to link to the corresponding article/info on how to add them

File: aspnet/fundamentals/environments.rst [hunk 2]
```diff
@@ -72,6 +72,24 @@ For example, you can use the following code in you Configure method to setup env
...
+  </environment>
+  <environment names="Staging,Production">
+    <link rel="stylesheet" href="https://ajax.aspnetcdn.com/ajax/bootstrap/3.3.5/css/bootstrap.min.css"
+      asp-fallback-href="~/lib/bootstrap/dist/css/bootstrap.min.css"
+      asp-fallback-test-class="sr-only" asp-fallback-test-property="position"
+      asp-fallback-test-value="absolute" />
+    <link rel="stylesheet" href="~/css/site.min.css" asp-append-version="true" />
+  </environment>
+
+You can define environmental settings for use in Tag Helpers in the project’s debug properties or the ``launchSettings.json`` file.
```
Comment (dougbu) [inline aspnet/fundamentals/environments.rst]: Not sure about focusing on the `launchSettings.json` file. The system has many ways to set the current environment and `launchSettings.json` is just as VS-specific as "the project’s debug properties".

Suggest instead mentioning two distinct options such as "set the `ASPNETCORE_ENVIRONMENT` environment variable at the command prompt (before starting a self-hosted site) or change the same variable in Visual Studio -- See the `Environment Variables` section of the project's Debug properties tab." (Note I'm not sure how to correctly reference that tab.)

**Comment by Rick-Anderson** [comment]
https://docs.asp.net/en/latest/mvc/views/tag-helpers/intro.html shows how to include tag helpers.

**Comment by danroth27** [comment]
:up::date:

File: aspnet/fundamentals/environments.rst [hunk 3]
```diff
@@ -72,6 +72,15 @@ For example, you can use the following code in you Configure method to setup env
...
 If the app is running in a ``Development`` environment, then it enables BrowserLink and development specific error pages (which typically should not be run in production). Otherwise, if the app is not running in a development environment, a standard error handling page is configured to be displayed in response to any unhandled exceptions.
 
+You may need to determine which content to send to the client at runtime, depending on the current environment. For example, in a development environment you generally serve non-minimized scripts and style sheets, which makes debugging easier. Production and test environments should serve the minified versions and generally from a CDN. You can do this using the Environment :doc:`tag helper </mvc/views/tag-helpers/intro>`. The Environment tag helper will only render its contents if the current environment matches one of the environments specified using the ``names`` attribute.
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/Views/Shared/_Layout.cshtml
+  :language: html
+  :lines: 8-17
+  :dedent: 8
+
+To get started with using Tag Helpers in your application see :doc:`/mvc/views/tag-helpers/intro`.
```
Comment (NTaylorMullen) [inline aspnet/fundamentals/environments.rst]: above you mention tag helpers lower case, should be consistent.

**Comment by danroth27** [comment]
:up::date:

**Comment by NTaylorMullen** [comment]
:shipit: if you give me $5.

**Comment by dougbu** [comment]
I'm ready to start a bidding war: :shipit: for $3.

**Pull request merged**